frontend/cypress/integration/common/service_details.ts

```diff
@@ -13,85 +13,86 @@ Given('a service in the cluster with a healthy amount of traffic', function () {
     openTab('Traffic');
-    cy.contains('Inbound Traffic');
-    cy.contains('No Inbound Traffic').should('not.exist');
-    cy.contains('Outbound Traffic');
-    cy.contains('No Inbound Traffic').should('not.exist');
-    cy.contains('.pf-c-card__body').within(() => {
+    cy.get('.pf-c-card__body').within(() => {
+        cy.contains('Inbound Traffic');
+        cy.contains('No Inbound Traffic').should('not.exist');
+        cy.contains('Outbound Traffic');
+        cy.contains('No Inbound Traffic').should('not.exist');
         cy.get('table.pf-c-table.pf-m-grid-md').should('exist');
         cy.contains('istio-ingressgateway');
     });
 });
 
-Then('the user sees {string} graph', (graph: string) => {
+Then('sd::user sees {string} graph', (graph: string) => {
     openTab('Inbound Metrics');
     cy.get('.pf-l-grid__item').children().children().children().contains(graph);
 });
 
-Then('the user does not see No data message in the {string} graph', (graph: string) => {
+Then('sd::user does not see No data message in the {string} graph', (graph: string) => {
     openTab('Inbound Metrics');
     cy.get('.pf-l-grid__item').children().children().children().contains(graph).should('not.contain', 'No data available');
 });
 
-Then('user sees graph with traces information', (graph: string) => {
+Then('sd::user sees graph with traces information', (graph: string) => {
     openTab('Traces');
     cy.getBySel('jaeger-scatterplot');
     cy.getBySel('trace-details-tabs').should('not.exist');
     cy.getBySel('jaeger-scatterplot').contains('Traces');
     cy.getBySel('jaeger-scatterplot').find(`path${tracingDotQuery}`).first().should('be.visible').click({ force: true });
 });
 
-And('And user sees trace details after selecting a trace', () => {
+And('sd::user sees trace details after selecting a trace', () => {
     cy.getBySel('trace-details-tabs').should('be.visible').contains("Trace Details");
-    cy.getBySel('.pf-l-grid').should('be.visible').get(".pf-c-label__content").should('be.visible');
     cy.getBySel('trace-details-kebab').click().contains('View on Graph');
 });
 
-And('user sees table details after selecting a trace', () => {
+And('sd::user sees table details after selecting a trace', () => {
     cy.getBySel('trace-details-tabs').should('be.visible').contains("Span Details").click();
-    cy.get('table').should('be.visible').contains('th', 'Timeline');
-    cy.get('#pf-tab-section-1-trace-details').within(() => {
-        cy.get('.pf-c-dropdown__toggle.pf-m-plain').first().should('be.visible').click();
-        cy.get('ul.pf-c-dropdown__menu').should('be.visible').contains('Inbound Metrics');
-    });
+    cy.get('table').should('be.visible')
+      .find('tbody tr')                // ignore thead rows
+      .should('have.length.above', 1)  // retries above cy.find() until we have a non head-row
+      .eq(1)                           // take 1st  row
+      .find('td').eq(4)                // take 5th cell (kebab)
+      .click()
```

Comment:
There actually are some helper functions for interacting with tables in [table.ts](https://github.com/kiali/kiali/blob/master/frontend/cypress/integration/common/table.ts#L12-L47) but if this works too or is specific to this test then that's fine.

Reply:
Ah, good to know. Thanks.